services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/attachment/hanke/HankeAttachmentMigratorITest.kt

```diff
@@ -0,0 +1,161 @@
+package fi.hel.haitaton.hanke.attachment.hanke
+
+import assertk.all
+import assertk.assertThat
+import assertk.assertions.isEmpty
+import assertk.assertions.isEqualTo
+import assertk.assertions.isNotNull
+import assertk.assertions.isNull
+import assertk.assertions.prop
+import fi.hel.haitaton.hanke.DatabaseTest
+import fi.hel.haitaton.hanke.attachment.DEFAULT_DATA
+import fi.hel.haitaton.hanke.attachment.DUMMY_DATA
+import fi.hel.haitaton.hanke.attachment.USERNAME
+import fi.hel.haitaton.hanke.attachment.azure.Container.HANKE_LIITTEET
+import fi.hel.haitaton.hanke.attachment.common.AttachmentContent
+import fi.hel.haitaton.hanke.attachment.common.AttachmentValidator.parseMediaType
+import fi.hel.haitaton.hanke.attachment.common.DownloadResponse
+import fi.hel.haitaton.hanke.attachment.common.HankeAttachmentEntity
+import fi.hel.haitaton.hanke.attachment.common.MigrationResult
+import fi.hel.haitaton.hanke.attachment.common.MockFileClient
+import fi.hel.haitaton.hanke.attachment.common.MockFileClientExtension
+import fi.hel.haitaton.hanke.attachment.common.UnMigratedHankeAttachment
+import fi.hel.haitaton.hanke.factory.AttachmentFactory
+import fi.hel.haitaton.hanke.factory.HankeAttachmentFactory
+import fi.hel.haitaton.hanke.test.Asserts.isValidBlobLocation
+import java.util.UUID
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.security.test.context.support.WithMockUser
+import org.springframework.test.context.ActiveProfiles
+
+@SpringBootTest
+@ActiveProfiles("test")
+@WithMockUser(USERNAME)
+@ExtendWith(MockFileClientExtension::class)
+class HankeAttachmentMigratorITest(
+    @Autowired private val migrator: HankeAttachmentMigrator,
+    @Autowired private val attachmentFactory: HankeAttachmentFactory,
+    @Autowired private val fileClient: MockFileClient
+) : DatabaseTest() {
+
+    @Nested
+    inner class MigrationOperation {
+        @Test
+        fun `Should migrate to cloud and remove old content`() {
```

Comment:
This test seems to test the `migrate()`-function in the test file that emulates the operation that the scheduler performs. There are no guarantees that the emulation will match the actual implementation. Why not test the scheduler function directly (in a separate test file)?

Reply:
Done.